### ISSUE TITLE
Cannot communicate Kubernetes API Server in EKS start 1.14

### ISSUE THREAD

**Issue by DingGGu** [cilium/cilium#28882]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Start Cilium 1.14, Certain pods cannot connect to control plane.
You know, EKS provider control plane host 172.20.0.1:443.

Example of cluster-autoscaler's log:
```
F1031 04:36:45.526866       1 main.go:508] Failed to get nodes from apiserver: Get "https://172.20.0.1:443/api/v1/nodes": dial tcp 172.20.0.1:443: i/o timeout
```

1. Symptoms of cannot connect to with kube-apiserver occur on a node-by-node basis. When one of the nodes becomes unable to communicate and is “contagious” to other nodes when it is drain problem node.
2. Restart problem pod can solve error, not always. However the symptom recurs 2 to 5 days later.
3. I confirmed that there was no problem in 1.13.
4. I confirmed that the same problem occurs in EKS 1.26 to 1.28.
5. I confirmed that the same problem occurs in Cilium 1.14.0 ~ 1.14.3

### Cilium Version

```
Client: 1.14.3 252a99ef 2023-10-18T18:21:56+03:00 go version go1.20.10 linux/amd64
Daemon: 1.14.3 252a99ef 2023-10-18T18:21:56+03:00 go version go1.20.10 linux/amd64
```

### Kernel Version

```
Linux ip-10-74-107-86 5.15.0-1045-aws #50~20.04.1-Ubuntu SMP Wed Sep 6 17:29:11 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

### Kubernetes Version

```
Server Version: version.Info{Major:"1", Minor:"26+", GitVersion:"v1.26.9-eks-f8587cb", GitCommit:"2e0997b730999ec918be56b8ed945394905673c5", GitTreeState:"clean", BuildDate:"2023-09-18T22:09:26Z", GoVersion:"go1.20.8", Compiler:"gc", Platform:"linux/amd64"}
```

### Sysdump

I can't provide sysdump to public. Please understand.

```
# vaules.yaml for helm
k8sServiceHost: <<EKS_API_SERVER_HOST>>
k8sServicePort: 443

eni:
  enabled: true
ipam:
  mode: eni
tunnel: disabled
kubeProxyReplacement: true
bpf:
  masquerade: true
loadBalancer:
  serviceTopology: true

cni:
  exclusive: false # for istio

envoy:
  enabled: true
```

### Anything else?

Somebody get same error. you can find in comment reaction.
https://github.com/cilium/cilium/issues/20679#issuecomment-1677199340

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by DingGGu** [comment]
With similar symptoms, some Pods have an issue where DNS Resolving fails.

Once a Pod had a problem, DNS Resolving will continue to fail until the Pod is deleted.
(Because there is a DNS Resolving probe in the container, the pod is not deleted and the container is restarted)

Result of tcpdump,
For Pods that are operating normally on the same node, packets are delivered from pod-ip:53 -> core-dns:53.
However, the problematic Pod does not receive a response during DNS Resolving, and no packets are recorded in tcpdump.

This appears to indicate that the Pod's network is not configured correctly.

Here is a cilium agent that node debuginfo.

```
$ ip route
default via 10.91.64.1 dev ens5 proto dhcp src 10.91.87.162 metric 100
10.91.64.1 dev ens5 proto dhcp scope link src 10.91.87.162 metric 100
10.91.64.172 dev lxc8ffa81e75751 proto kernel scope link
```

#### BPF Policy Get 1173

```
POLICY   DIRECTION   IDENTITY   PORT/PROTO   PROXY PORT   AUTH TYPE   BYTES    PACKETS   PREFIX
Allow    Ingress     0          ANY          NONE         disabled    0        0         0
Allow    Ingress     1          ANY          NONE         disabled    101010   1365      0
Allow    Egress      0          ANY          NONE         disabled    336768   2989      0

```


#### BPF CT List 1173

```
Invalid argument: unknown type 1173
```


#### Endpoint Get 1173

```
[
  {
    "id": 1173,
    "spec": {
      "label-configuration": {},
      "options": {
        "ConntrackAccounting": "Enabled",
        "ConntrackLocal": "Disabled",
        "Debug": "Disabled",
        "DebugLB": "Disabled",
        "DebugPolicy": "Disabled",
        "DropNotification": "Enabled",
        "MonitorAggregationLevel": "Medium",
        "PolicyAuditMode": "Disabled",
        "PolicyVerdictNotification": "Enabled",
        "SourceIPVerification": "Enabled",
        "TraceNotification": "Enabled"
      }
    },
    "status": {
      "controllers": [
        {
          "configuration": {
            "error-retry": true,
            "error-retry-base": "2s"
          },
          "name": "endpoint-1173-regeneration-recovery",
          "status": {
            "last-failure-timestamp": "0001-01-01T00:00:00.000Z",
            "last-success-timestamp": "0001-01-01T00:00:00.000Z"
          },
          "uuid": "3389c77c-2816-4d0c-bbfe-ca154ac8a43e"
        },
        {
          "configuration": {
            "error-retry": true,
            "interval": "5m0s"
          },
          "name": "resolve-identity-1173",
          "status": {
            "last-failure-timestamp": "0001-01-01T00:00:00.000Z",
            "last-success-timestamp": "2023-10-31T12:00:15.534Z",
            "success-count": 9
          },
          "uuid": "36365d09-91f8-4eee-a2bb-6f09c57a4812"
        },
        {
          "configuration": {
            "error-retry": true,
            "interval": "15m0s"
          },
          "name": "sync-policymap-1173",
          "status": {
            "last-failure-timestamp": "0001-01-01T00:00:00.000Z",
            "last-success-timestamp": "2023-10-31T11:50:15.715Z",
            "success-count": 3
          },
          "uuid": "3a1726a3-58b0-45cc-b678-2ceb54420e52"
        },
        {
          "configuration": {
            "error-retry": true,
            "interval": "10s"
          },
          "name": "sync-to-k8s-ciliumendpoint (1173)",
          "status": {
            "last-failure-timestamp": "0001-01-01T00:00:00.000Z",
            "last-success-timestamp": "2023-10-31T12:01:35.725Z",
            "success-count": 250
          },
          "uuid": "b6a9c698-f6c5-4f66-899a-98b27bed3938"
        }
      ],
      "external-identifiers": {
        "container-id": "...",
        "k8s-namespace": "...",
        "k8s-pod-name": "...",
        "pod-name": ".../..."
      },
      "health": {
        "bpf": "OK",
        "connected": true,
        "overallHealth": "OK",
        "policy": "OK"
      },
      "identity": {
        "id": 273698,
        "labels": [
          ...
        ]
      },
      "labels": {
        "derived": [
          "k8s:pod-template-hash=..."
        ],
        "realized": {},
        "security-relevant": [
...
        ]
      },
      "log": [
        {
          "code": "OK",
          "message": "Successfully regenerated endpoint program (Reason: one or more identities created or deleted)",
          "state": "ready",
          "timestamp": "2023-10-31T12:00:01Z"
        }
      ],
      "namedPorts": [
        {
          "name": "http",
          "port": 3000,
          "protocol": "TCP"
        },
        {
          "name": "http-envoy-prom",
          "port": 15090,
          "protocol": "TCP"
        }
      ],
      "networking": {
        "addressing": [
          {
            "ipv4": "10.91.64.172",
            "ipv4-pool-name": "default"
          }
        ],
        "host-mac": "...",
        "interface-index": 2465,
        "interface-name": "lxc8ffa81e75751",
        "mac": "..."
      },
      "policy": {
        "proxy-statistics": [],
        "realized": {
          "allowed-egress-identities": [
            0
          ],
          "allowed-ingress-identities": [
            0,
            1
          ],
          "build": 2,
          "denied-egress-identities": null,
          "denied-ingress-identities": null,
          "id": 273698,
          "l4": {
            "egress": [],
            "ingress": []
          },
          "policy-enabled": "none",
          "policy-revision": 2
        },
        "spec": {
          "allowed-egress-identities": [
            0
          ],
          "allowed-ingress-identities": [
            0,
            1
          ],
          "build": 2,
          "denied-egress-identities": null,
          "denied-ingress-identities": null,
          "id": 273698,
          "l4": {
            "egress": [],
            "ingress": []
          },
          "policy-enabled": "none",
          "policy-revision": 2
        }
      },
      "realized": {
        "label-configuration": {},
        "options": {
          "ConntrackAccounting": "Enabled",
          "ConntrackLocal": "Disabled",
          "Debug": "Disabled",
          "DebugLB": "Disabled",
          "DebugPolicy": "Disabled",
          "DropNotification": "Enabled",
          "MonitorAggregationLevel": "Medium",
          "PolicyAuditMode": "Disabled",
          "PolicyVerdictNotification": "Enabled",
          "SourceIPVerification": "Enabled",
          "TraceNotification": "Enabled"
        }
      },
      "state": "ready"
    }
  }
]


```


#### Endpoint Health 1173

```
Overall Health:   OK
BPF Health:       OK
Policy Health:    OK
Connected:        yes

```


#### Endpoint Log 1173

```
Timestamp              Status   State                   Message
2023-10-31T12:00:01Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T12:00:01Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T12:00:01Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T12:00:01Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T12:00:00Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T12:00:00Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T12:00:00Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T12:00:00Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:55:01Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:55:01Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:55:01Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:55:01Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:55:00Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:55:00Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:55:00Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:55:00Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:50:01Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:50:01Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:50:01Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:50:01Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:50:00Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:50:00Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:50:00Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:50:00Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:46:50Z   OK       ready                   Successfully regenerated endpoint program (Reason: )
2023-10-31T11:46:50Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:46:50Z   OK       regenerating            Regenerating endpoint:
2023-10-31T11:46:50Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to
2023-10-31T11:46:49Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:46:49Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:46:49Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:46:49Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:46:48Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:46:48Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:46:48Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:46:48Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:46:47Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:46:47Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:46:47Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:46:47Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:45:01Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:45:01Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:45:01Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:45:01Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:45:00Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:45:00Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:45:00Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:45:00Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:43:36Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:43:36Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:43:36Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:43:36Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:40:55Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:40:55Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:40:55Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:40:55Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:40:01Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:40:01Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:40:01Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:40:01Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:40:00Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:40:00Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:40:00Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:40:00Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:38:49Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:38:49Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:38:49Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:38:49Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:38:20Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:38:20Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:38:20Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:38:20Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:37:43Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:37:43Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:37:43Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:37:43Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:36:00Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:36:00Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:36:00Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:36:00Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:35:01Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:35:01Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:35:01Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:35:01Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:35:00Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:35:00Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:35:00Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:35:00Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:31:49Z   OK       ready                   Successfully regenerated endpoint program (Reason: )
2023-10-31T11:31:49Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:31:49Z   OK       regenerating            Regenerating endpoint:
2023-10-31T11:31:49Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to
2023-10-31T11:31:48Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:31:48Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:31:48Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:31:48Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:31:47Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:31:47Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:31:47Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:31:47Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:31:46Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:31:46Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:31:46Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:31:46Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:30:01Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:30:01Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:30:01Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:30:01Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:30:00Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:30:00Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:30:00Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:30:00Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:29:55Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:29:55Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:29:55Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:29:55Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:29:28Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:29:28Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:29:28Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:29:28Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:25:01Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:25:01Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:25:01Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:25:01Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:25:00Z   OK       ready                   Successfully regenerated endpoint program (Reason: one or more identities created or deleted)
2023-10-31T11:25:00Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:25:00Z   OK       regenerating            Regenerating endpoint: one or more identities created or deleted
2023-10-31T11:25:00Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to one or more identities created or deleted
2023-10-31T11:20:15Z   OK       ready                   Successfully regenerated endpoint program (Reason: updated security labels)
2023-10-31T11:20:15Z   OK       ready                   Completed endpoint regeneration with no pending regeneration requests
2023-10-31T11:20:15Z   OK       regenerating            Regenerating endpoint: updated security labels
2023-10-31T11:20:15Z   OK       waiting-to-regenerate   Triggering endpoint regeneration due to updated security labels
2023-10-31T11:20:15Z   OK       ready                   Set identity for this endpoint
2023-10-31T11:20:15Z   OK       waiting-for-identity    Endpoint creation

```


#### Identity get 273698

```
ID       LABELS
273698   ...

```

**Comment by oarezki** [comment]
I run into the same issue but it was while adding new nodes.
I figured out that it's kube proxy's absence which caused that.
I had to set it up again, at least temporarily.

**Comment by DingGGu** [comment]
> I run into the same issue but it was while adding new nodes. I figured out that it's kube proxy's absence which caused that. I had to set it up again, at least temporarily.

Did disabling kubeProxyReplacement work?

**Comment by oarezki** [comment]
I left kubeProxyReplacement as it was (to true) as I needed Cilium to handle kube proxy role.
However, since Cilium documentation suggested that one can/has to delete kube proxy ds, then I did so.
After encountering my [issue](https://github.com/cilium/cilium/issues/28807), I understood that I had to add kube proxy again, so cilium agents can communicate with API server.

**Comment by DingGGu** [comment]
`kubeProxyReplacement: true` does not need kube-proxy.

The [documents](https://docs.cilium.io/en/stable/network/kubernetes/kubeproxy-free/) guide to delete kube-proxy daemonset.

I think that's not seem to be the normal direction.

**Comment by oarezki** [comment]
Did you also delete it ? If so, did you attempt to add it again to see if it solves (as a workaround) the issue ?

**Comment by DingGGu** [comment]
Since we expect the fastest networking through BPF, we have removed KubeProxy. If a problem occurs next time, I will revive kube-proxy and check. but, iptables is created when kube-proxy is enabled, and cilium and kube-proxy are responsible for routing at the same time, so this does not seem to be a normal method.

**Comment by DingGGu** [comment]
If you downgrade to 1.13, it generally works very well, so please check it out. However, if you use 1.13, you cannot use 100% BPF Routing mode in ENI. Starting from 1.14, ENI's BPF mode was supported.

**Comment by oarezki** [comment]
I get your concerns. However, you won't keep kube proxy, I **only add it the time** the nodes are added.

If kubeProxyReplacement is set to true, even if there is kube proxy, cilium will still be fully handling kube proxy role.

When it is to true, you get all the below services types enabled and handled by Cilium. 

```
Services:
  - ClusterIP:      Enabled
  - NodePort:       Enabled (Range: 30000-32767)
  - LoadBalancer:   Enabled
  - externalIPs:    Enabled
  - HostPort:       Enabled
```

```
if option.Config.KubeProxyReplacement == option.KubeProxyReplacementStrict ||
		option.Config.KubeProxyReplacement == option.KubeProxyReplacementTrue {

		log.Infof("Auto-enabling %q, %q, %q, %q, %q features",
			option.EnableNodePort, option.EnableExternalIPs,
			option.EnableSocketLB, option.EnableHostPort,
			option.EnableSessionAffinity)

		option.Config.EnableHostPort = true
		option.Config.EnableNodePort = true
		option.Config.EnableExternalIPs = true
		option.Config.EnableSocketLB = true
		option.Config.EnableSessionAffinity = true
```
Anyway, let's hope this gets clarified.

**Comment by DingGGu** [comment]
Oh, so you mean that when the node starts, kube-proxy exists for a short time, and then when cilium is ready, kube-proxy is removed and iptables is also cleaned up?

**Comment by DingGGu** [comment]
Maybe related or duplicate issue of https://github.com/cilium/cilium/issues/27343

**Comment by oarezki** [comment]
Yes. kube proxy ds should be geared up before scaling out process is initiated, you add a new node, there will consequently be a new kube proxy pod scheduled on that node, which seems to be the key for a success connection with API server, and which cilium agent, config init container needs to go on. Once, it's done, kube proxy ds can be deleted, without any issue.

Regarding [#27343](https://github.com/cilium/cilium/issues/27343), I think it's related. The issue with **hostLegacyRouting** set to true (which looks like a workaround for some) is that **one won't benefit all of bpf efficiency**.

As this issue may appear in different circumstances, if it's about scaling out, I'd prefer adding kube proxy temporarily than **hostLegacyRouting: true**.

**Comment by DingGGu** [comment]
Resolved Pr was not merged in v1.14.4. waiting v1.14.5

**Comment by project-administrator** [comment]
@DingGGu Is there any PR open which resolves this issue?

**Comment by DingGGu** [comment]
https://github.com/cilium/cilium/pull/28264 is the pr which was merged in v1.14.5 on today.

**Comment by julianwiedmann** [comment]
> Resolved Pr was not merged in v1.14.4. waiting v1.14.5

:wave: hi @DingGGu , just checking - did `v1.14.5` end up resolving this issue for you?

**Comment by DingGGu** [comment]
> > Resolved Pr was not merged in v1.14.4. waiting v1.14.5
> 
> 👋 hi @DingGGu , just checking - did `v1.14.5` end up resolving this issue for you?

Hi, Issue was resolved in v1.14.5
